src/Panel.vue

```diff
@@ -160,92 +176,78 @@
         if (this.type !== 'seamless') {
-          return !this.noClose;
+          return !this.noCloseBool;
         } else {
           return onHeaderHover;
         }
       }
     },
+    data: function () {
+      return {
+        onHeaderHover: false,
+        localMinimized: '',
+        localExpanded: ''
+      }
+    },
     methods: {
-      toggle () {
-        this.expanded = !this.expanded
+      toggle() {
+        this.localExpanded = !this.localExpanded;
       },
       expand() {
-        if (this.expanded) {
-          // Ask children to collapse
-          this.$broadcast('panel:collapse', this.ctrlLvl)
-        } else {
-          // Expand children
-          this.$broadcast('panel:expand', this.ctrlLvl)
-        }
-        this.expanded = !this.expanded
+        this.localExpanded = !this.localExpanded;
       },
       close() {
-        console.log('close')
-        this.minimized = true;
-      },
-      open () {
-        this.expanded = true;
-        this.minimized = false;
-      },
-      expandCollapseHandler (isExpand, level) {
-        if (level > 0) {
-          this.canCollapse && (this.expanded = isExpand)
-          this.$broadcast('panel:' + (isExpand ? 'expand' : 'collapse'), level - 1)
-        } else if (level === -1) {
-          this.canCollapse && (this.expanded = isExpand)
-          this.$broadcast('panel:' + (isExpand ? 'expand' : 'collapse'), -1)
-        }
+        this.localMinimized = true;
+      },
+      open() {
+        this.localExpanded = true;
+        this.localMinimized = false;
       },
       scrollIntoViewIfNeeded() {
-        var top = this.$el.getBoundingClientRect().top;
-        var isTopInView = (top >= 0) && (top <= window.innerHeight);
+        let top = this.$el.getBoundingClientRect().top;
+        let isTopInView = (top >= 0) && (top <= window.innerHeight);
         if (!isTopInView) {
           this.$el.scrollIntoView();
         }
       },
       collapseThenScrollIntoViewIfNeeded() {
-        this.$once('isOpenEvent', (el, isOpen) => {
+        this.$once('is-open-event', (el, isOpen) => {
           this.scrollIntoViewIfNeeded();
         });
         this.expand();
       },
       openPopup() {
         window.open(this.popupUrl);
+      },
+      retrieveOnOpen(el, isOpen) {
+        if (isOpen && this.isDynamic) {
+          this.$refs.retriever.fetch()
+        }
       }
     },
     watch: {
-      'expanded': function (val, oldVal) {
-        this.$dispatch('isOpenEvent', this, val)
-      }
-    },
-    events: {
-      'panel:expand': function (level) {
-        this.expandCollapseHandler(true, level)
+      'localExpanded': function (val, oldVal) {
+        this.retrieveOnOpen(this, val);
       },
-      'panel:collapse': function (level) {
-        this.expandCollapseHandler(false, level)
-      }
     },
     created () {
-      if (this.expanded === null) {
-        this.expanded = !this.canCollapse
+      if (this.expandedBool === null) {
```

Comment:
This will never evaluate to true, because `expandedBool()` will never be null.

Using the raw variable `this.expanded` should be fine here.

Reply:
Since I'm setting `this.localExpanded` below, I can remove the if statement block entirely.